src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,74 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+type Props = {
+  onAddMovie: (movie: Movie) => void;
+};
+
+export const FindMovie: React.FC<Props> = ({ onAddMovie }) => {
+  const [searchValue, setSearchValue] = useState('');
+  const [foundMovie, setFoundMovie] = useState<Movie | null>(null);
+  const [error, setError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [firstFound, setFirstFound] = useState(false);
+
+  const handleFindMovie = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    getMovie(searchValue)
+      .then((movieData) => {
+        if (movieData.Response === 'True') {
+          const {
+            Title: title,
+            Plot: description,
+            Poster,
+            imdbID: imdbId,
+          } = movieData;
+
+          const imgUrl = Poster === 'N/A'
+            ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+            : Poster;
+
+          const imdbUrl = `https://www.imdb.com/title/${imdbId}`;
+
+          setFoundMovie({
+            title,
+            description,
+            imgUrl,
+            imdbUrl,
+            imdbId,
+          });
+          setFirstFound(true);
+        }
+
+        if (movieData.Response === 'False') {
+          setError(true);
+        }
+
+        setIsLoading(false);
+      });
+  };
+
+  const handleSearchInput = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(event.target.value);
+    setError(false);
+  };
+
+  const handleAddMovieButton = () => {
+    onAddMovie(foundMovie as Movie);
```

Comment:
```suggestion
    onAddMovie(foundMovie as Movie);
```
it would be better to add check if foundMovie is not null